### PR TITLE
Enable JSF 2.3 config tests for Jakarta EE10 Faces 4.0

### PR DESCRIPTION
**Pull request by pnicolucci**
for #22368

### CODE REVIEW

**Comment by pnicolucci** [comment]
#libby #build

**Comment by pnicolucci** [comment]
#libby #build

File: dev/com.ibm.ws.jsf.2.3_fat/test-applications/CDICommon/src/com/ibm/ws/jsf23/fat/cdi/common/managed/CustomStateManager.java [hunk 1]
```diff
@@ -86,16 +83,4 @@ public boolean isSavingStateInClient(FacesContext context) {
...
         return super.isSavingStateInClient(context);
     }
-
-    /*
-     * (non-Javadoc)
-     *
-     * @see javax.faces.application.StateManagerWrapper#getWrapped()
-     */
-    @Override
-    public StateManager getWrapped() {
```
Comment (volosied) [inline dev/com.ibm.ws.jsf.2.3_fat/test-applications/CDICommon/src/com/ibm/ws/jsf23/fat/cdi/common/managed/CustomStateManager.java:96]: Why was this removed? Unnecessary for the test?

Comment (volosied) [reply dev/com.ibm.ws.jsf.2.3_fat/test-applications/CDICommon/src/com/ibm/ws/jsf23/fat/cdi/common/managed/CustomStateManager.java:96]: Ah, super call

File: dev/com.ibm.ws.jsf.2.3_fat/test-applications/CDICommon/src/com/ibm/ws/jsf23/fat/cdi/common/managed/CustomELResolver.java [hunk 2]
```diff
@@ -85,9 +93,17 @@ public Object getValue(ELContext context, Object base, Object property) {
                 else
                     outcome += _methodBean.getData();
 
-                ((TestCustomBean) base).setData(outcome);
+                if (className40Found) {
+                    ((com.ibm.ws.jsf23.fat.cdi.common.beans.faces40.TestCustomBean) base).setData(outcome);
+                } else {
+                    ((TestCustomBean) base).setData(outcome);
+                }
```
Comment (volosied) [inline dev/com.ibm.ws.jsf.2.3_fat/test-applications/CDICommon/src/com/ibm/ws/jsf23/fat/cdi/common/managed/CustomELResolver.java:101]: Small style thing: you have curly brackets for these if statements but not above.

File: dev/com.ibm.ws.jsf.2.3_fat/fat/src/com/ibm/ws/jsf23/fat/tests/JSF23CDIConfigByACPTests.java [hunk 3]
```diff
@@ -47,7 +47,13 @@ public static void setup() throws Exception {
 
         // Create the CDIConfigByACP jar that is used in CDIConfigByACP.war,
         JavaArchive cdiConfigByACPJar = ShrinkWrap.create(JavaArchive.class, "CDIConfigByACP.jar");
-        cdiConfigByACPJar.addPackage("com.ibm.ws.jsf23.fat.cdi.common.beans");
+        if (JakartaEE10Action.isActive()) {
+            // Include @Named beans.
+            cdiConfigByACPJar.addPackage("com.ibm.ws.jsf23.fat.cdi.common.beans.faces40");
+        } else {
+            // Include @ManagedBean beans.
+            cdiConfigByACPJar.addPackage("com.ibm.ws.jsf23.fat.cdi.common.beans");
```
Comment (volosied) [inline dev/com.ibm.ws.jsf.2.3_fat/fat/src/com/ibm/ws/jsf23/fat/tests/JSF23CDIConfigByACPTests.java:55]: I think it would be nice to have a separate package for the 2.3  beans (rather than have beans under the common.beans package and then also including the faces40 package within), but I understand that would be more refactoring. 

I think it's fine as is.

**Review by volosied — APPROVED** [review]
Looks fine to me. Let's see how the build goes.

File: dev/com.ibm.ws.jsf.2.3_fat/fat/src/com/ibm/ws/jsf23/fat/tests/JSF23CDIConfigByACPTests.java (hunk 3 above)
Comment (pnicolucci) [reply dev/com.ibm.ws.jsf.2.3_fat/fat/src/com/ibm/ws/jsf23/fat/tests/JSF23CDIConfigByACPTests.java:55]: I fixed it! Thanks!

**Comment by pnicolucci** [comment]
#libby #build

File: dev/com.ibm.ws.jsf.2.3_fat/test-applications/CDICommon/src/com/ibm/ws/jsf23/fat/cdi/common/managed/CustomELResolver.java (hunk 2 above)
Comment (pnicolucci) [reply dev/com.ibm.ws.jsf.2.3_fat/test-applications/CDICommon/src/com/ibm/ws/jsf23/fat/cdi/common/managed/CustomELResolver.java:101]: I fixed it thanks!
